### PR TITLE
[risk=low] Databrowser measurement query and api update

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Update measurement count generation queries to generate histogram data for each unit.
2. Updated ui and api for the same

### CODE REVIEW

File: api/.Rhistory [hunk 1]
```diff
@@ -0,0 +1,2 @@
+exit
```
Comment (biopete) [inline api/.Rhistory]: What is this ?

File: api/db-cdr/generate-cdr/bq-schemas/unit_map.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  {
+    "mode":"REQUIRED",
+    "name": "unit_concept_id",
+    "type": "INTEGER"
+  },
+  {
+    "mode":"REQUIRED",
+    "name": "unit_source_value",
+    "type": "STRING"
+  }
```
Comment (biopete) [inline api/db-cdr/generate-cdr/bq-schemas/unit_map.json]: For later to think about --  I'm tempted to add a preferred_name flag here that would flag this unit_source_value as the preferred name.  But that is another story.

File: public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts [hunk 3]
```diff
@@ -112,4 +125,10 @@ export class ConceptChartsComponent implements OnInit, OnDestroy {
       analysis.results.push(this.otherGenderResult);
     }
   }
+
+    showHistogram(unit: string) {
```
Comment (biopete) [inline public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts]: This code in this function deals with just the gender analysis? Can you make a better name for it. And reuse it above where  this code is duplicated?

File: public-ui/src/app/views/pm/pm.component.html [hunk 4]
```diff
@@ -41,20 +41,20 @@ <h5 class="box-title"> {{selectedGroup.groupName}}</h5>
                          selectedConcept.analyses.measurementValueGenderAnalysis" [genderId]="dbc.FEMALE_GENDER_ID"
                      [chartTitle]="gender.analysisStratumName + ' - ' + gender.countValue.toLocaleString()"
                      [chartType]="selectedConcept.chartType"
-                     [analysis]="selectedConcept.analyses.measurementValueGenderAnalysis"></app-chart>
+                     [analysis]="selectedConcept.analyses.measurementValueGenderAnalysis[0]"></app-chart>
```
Comment (biopete) [inline public-ui/src/app/views/pm/pm.component.html]: So on this page, we are only showing the first one? That seems reasonable right now. Have you looked at the data in prod to see if all our ppi data comes in mapped to one unit or what ?

File: api/db-cdr/generate-cdr/make-bq-public-data.sh [hunk 5]
```diff
@@ -171,7 +171,36 @@ set count_value =
...
+        when count_value  > 0 and count_value >= ${BIN_SIZE}
+            then ROUND(CEIL(CEIL(count_value / ${BIN_SIZE}) * ${BIN_SIZE}/ ${person_count}), 2)
+        when source_count_value  > 0 and source_count_value < ${BIN_SIZE}
+            then ROUND(CEIL(${BIN_SIZE} / ${person_count}),2)
+        when source_count_value  > 0 and source_count_value >= ${BIN_SIZE}
+            then ROUND(CEIL(CEIL(source_count_value / ${BIN_SIZE}) * ${BIN_SIZE}/ ${person_count}), 2)
+        else
+            0.00
+    end
+where count_value = 0 and vocabulary_id='ppi' "
```
[the remark this answers is not in the document]
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/make-bq-public-data.sh]: Remember kayla asked to round ppi 0 concept counts to 20 ?

[the remark this answers is not in the document]
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/make-bq-public-data.sh]: Remember kayla asked to round up the counts that are 0 to 20 for ppi concepts.

File: api/.Rhistory (hunk 1 above)
Comment (SrushtiGangireddy) [reply api/.Rhistory]: sorry bad file removed it.

File: api/db-cdr/generate-cdr/bq-schemas/unit_map.json (hunk 2 above)
Comment (SrushtiGangireddy) [reply api/db-cdr/generate-cdr/bq-schemas/unit_map.json]: Good idea to try.

File: public-ui/src/app/views/pm/pm.component.html (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/views/pm/pm.component.html]: yeah, the ppi data except pregnany and wheel chair usage seems to have only one unit.

File: public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/concept-charts/concept-charts.component.ts]: Done.

File: api/db-cdr/generate-cdr/make-bq-public-data.sh [hunk 6]
```diff
@@ -161,36 +161,7 @@ set count_value =
...
-            then ROUND(CEIL(CEIL(count_value / ${BIN_SIZE}) * ${BIN_SIZE}/ ${person_count}), 2)
-        when source_count_value  > 0 and source_count_value < ${BIN_SIZE}
-            then ROUND(CEIL(${BIN_SIZE} / ${person_count}),2)
-        when source_count_value  > 0 and source_count_value >= ${BIN_SIZE}
-            then ROUND(CEIL(CEIL(source_count_value / ${BIN_SIZE}) * ${BIN_SIZE}/ ${person_count}), 2)
-        else
-            0.00
-    end
-where count_value = 0 and vocabulary_id='ppi' "
+where count_value > 0 or vocabulary_id='ppi' "
```
Comment (biopete) [inline api/db-cdr/generate-cdr/make-bq-public-data.sh]: is it 'PPI' or 'ppi' bq is case sensitive right ? Youcan just use lower

**Pull request merged**